### PR TITLE
Add requestContext as a lambda event property

### PR DESCRIPTION
**Pull request by DavidTanner**
context

When API Gateway creates a Lambda event it may also include a `requestContext` attribute.  We are using that in some projects to set the User, Account, Policy etc.  This makes It possible to set the attribute in the config for each request, and then is sent to Lambda.

https://docs.aws.amazon.com/lambda/latest/dg/services-apigateway.html
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-input.html
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html#context-variable-reference

### CODE REVIEW

**Comment by tqhoughton** [comment]
My understanding is that most of these values we map to header values. I know that this opens up a possible security vulnerability by doing it this way if someone tries to access a header in an unauthenticated endpoint, are there any other reasons we are adding this functionality?

**Review by tqhoughton — APPROVED** [review]

File: src/Alpha.d.ts [hunk 1]
```diff
@@ -14,7 +14,27 @@ export interface AlphaOptions extends AxiosRequestConfig {
   lambda?: Function
 }
 
-export type AlphaInstance = AxiosInstance;
+export interface AlphaRequestConfig extends AxiosRequestConfig {
+  requestContext?: Record<string, any>;
```
Comment (mdlavin) [inline src/Alpha.d.ts:18]: For non-lambda endpoints, will providing this attribute result in an error? I think it probably should, just to be safe

**Review by anthonyroach — CHANGES_REQUESTED** [review]
Is this really needed? We normally capture/pass most of the context information in headers (user, account, etc..) . Is there something missing?

File: src/Alpha.d.ts (hunk 1 above)
Comment (DavidTanner) [reply src/Alpha.d.ts:18]: As it is nothing will happen.

**Comment by DavidTanner** [comment]
@anthony.roach this is only applicable for instances that are invoked by API gateway directly, and only if they are checking the values in the event.

**Comment by DavidTanner** [comment]
@tqhoughton this isn't introducing a vulnerability, more line exposing it. If we are allowing someone to invoke the lambda directly then we are in big trouble.

This is only applicable for testing anyways.

**Comment by tqhoughton** [comment]
> @tqhoughton this isn't introducing a vulnerability, more line exposing it. If we are allowing someone to invoke the lambda directly then we are in big trouble.
> 
> This is only applicable for testing anyways.

Yeah that's what I meant, but I kind of agree with @anthonyroach that this change seems out of place, especially if it's only for testing purposes and serves no utility in its intended use. Is there a way to create API GW events with `requestContext` without using the alpha package that might be more suitable to your use case?

**Comment by DavidTanner** [comment]
It can be done manually, it is just so much nicer to use Alpha to create the body and headers.

**Comment by tqhoughton** [comment]
Maybe we could create a static `createEvent` method in this project that uses the same internal logic, but exposed for testing purposes?

**Pull request closed without merging**